### PR TITLE
NewHeader + NewFile

### PR DESCRIPTION
**Pull request by colinrgodsey**
Adding some missing but necessary constructors for File and Header types. It's now possible to create a FIT file from scratch!

-Colin
Engineering @ Ergatta, Inc.

### CODE REVIEW

**Comment by colinrgodsey** [comment]
This may also address #44

File: file.go [hunk 1]
```diff
@@ -53,6 +53,17 @@ type msgAdder interface {
 	add(reflect.Value)
 }
 
+// NewFile creates a new File of the given type.
+func NewFile(t FileType, h Header) (*File, error) {
+	f := new(File)
+	f.FileId.Type = t
+	f.Header = h
+	if err := f.init(); err != nil {
+		return nil, err
```
Comment (tormoder) [inline file.go:62]: Maybe:
```suggestion
		return nil, fmt.Errorf("error creating file: %w", err)
```

**Review by tormoder — COMMENTED** [review]
LGTM, thanks!

Would you mind squashing the commits?

**Comment by colinrgodsey** [comment]
fixedup, rebased, verified, and squashed!

**Review by tormoder — APPROVED** [review]

**Comment by davidkroell** [comment]
Hi @colinrgodsey, awesome work but how do i add `WeightScales` to the newly initialized `FitFile`?
In my project i created an issue about the re-implementation https://github.com/davidkroell/bodycomposition/issues/7.

Back then, i created #44 to add a function to set the Weight on a file, now we are able to create that file, but there is no functionality to add weight information to it? 
Am I missing something?

```go
header := fit.NewHeader(fit.V20, true)

newfile, err := fit.NewFile(fit.FileTypeWeight, header)
if err != nil {
	panic(err)
}

weight, err := newfile.Weight()
if err != nil {
	panic(err)
}

// weight.WeightScales  type: []*fit.WeightScaleMsg
```

We also need some method on the `FitFile` type, don't we?


Regards,
David

**Comment by tormoder** [comment]
I don't know if I understand you correctly, but you should be able to assign or append to the exported `weight.WeightScales` slice? `newfile.Weight()` gives you a reference/pointer you can manipulate.

**Pull request merged**